allennlp/nn/beam_search.py

```diff
@@ -197,8 +196,12 @@ def new_step(
         for key, state_tensor in state.items():
             if state_tensor is None:
                 continue
+            multilayer_rnn_decoder = state_tensor.dim() == 3 and key in {
```

Comment:
@dirkgr  thanks for your advice, it's more clearly. But I use `multilayer_rnn_decoder `  rather than `multilayer_decoder `, cause the original code seems works on `Transformer`.